release/src/main/groovy/mobilegaming-java-dataflow.groovy

```diff
@@ -110,7 +110,14 @@ class LeaderBoardRunner {
     }
     InjectorThread.stop()
     LeaderBoardThread.stop()
-    t.run("gcloud dataflow jobs cancel \$(gcloud dataflow jobs list | grep ${jobName} | grep Running | cut -d' ' -f1)")
+    t.run("""RUNNING_JOB=`gcloud dataflow jobs list | grep ${jobName} | grep Running | cut -d' ' -f1`
+if [ ! -z "\${RUNNING_JOB}" ] 
+  then 
+    gcloud dataflow jobs cancel \${RUNNING_JOB}
+  else 
+    echo "Job '${jobName}' is not running."
+fi 
+""")
```

Comment:
does it make sense to retry? are jobs here usually finished/cancelled, or pending?

Reply:
Jobs were finished/cancelled during my testing. I didn't see any in pending, there are a lot of sleeps/loops above but I suppose it could happen. We could search for 'Running\|Pending' if that makes sense.

I'm not sure a retry for cancelling will be that helpful here. If the job isn't present in the list as Running, or Pending, it won't need cancelling right?